index.js

```diff
@@ -1,95 +1,124 @@
+  thirstLevel -= 1
+  console.log("Ahh that hits the spot")
+  return thirstLevel
+}
+
+console.log(drinkWater(12))
+// => "Man I sure am thirsty"
+// => "Ahh that hits the spot"
+// => 11
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red") 
+function sameSameButDifferent (num, maybeNum) {
+  if (num === maybeNum) {
+    console.log("same same")
+  } else if (num === parseInt(maybeNum)){
+    console.log("same same (but different)")
+  } else {
+    console.log("different")
+  }
+}
+console.log(sameSameButDifferent(5, 5)) 
+// => "same same"
+
+console.log(sameSameButDifferent(123, "123")) 
+// => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 5)) 
-// // => "same same"
-//
-// console.log(sameSameButDifferent(123, "123")) 
-// // => "same same (but different)"
-//
-// console.log(sameSameButDifferent(5, 7)) 
-// // => "different"
-//
-// console.log(sameSameButDifferent(123, "122")) 
-// // => "different"
-// console.log("%c----------", "color: red") 
+console.log(sameSameButDifferent(5, 7)) 
+// => "different"
+
+console.log(sameSameButDifferent(123, "122")) 
+// => "different"
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)  
-// // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red") 
+console.log("%cQuestion 5", "color: red")
+const student1 = { name: "Duane", grade: 88 }
+function updateGrade(student, grade) {
+    student.grade = grade;
+}
+updateGrade(student1, 92)
+console.log(student1)  
+// => { name: "Duane", grade: 92 }
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 6 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
-//   }
-// ]
-// printNameAndPhones(users)
-// // => "Duane"
-// // => "Cell: 555-123-4567"
-// // => "Office: 555-456-7890"
-// // => "Liza"
-// // => "Cell: 555-234-5678"
-// // => "Office: 555-567-1234"
-// console.log("%c----------", "color: red") 
+console.log("%cQuestion 6", "color: red")
+
+const users = [ 
+  { 
+    name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
+  },
+  { 
+    name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
+  }
+]
+
+function printNameAndPhones(users) {
+  users.forEach( user_hash => {
+    console.log(user_hash['name']);
+    console.log(`Cell: ${user_hash['phones']['cell']}`);
+    console.log(`Office: ${user_hash['phones']['office']}`);
+  });
```

Comment:
Love to see the arrow function syntax! We'll go over that a bit later this week, but arrow functions are really nice for callbacks.

For accessing the properties of the user, it's more conventional in Javascript to use dot notation instead of bracket notation, and camelCase for variable names:

```js
  users.forEach( userObj => {
    console.log(userObj.name);
    console.log(`Cell: ${userObj.phones.cell}`);
    console.log(`Office: ${userObj.phones.office}`);
  });
```